src/modules/job-manager/annotate.c

```diff
@@ -167,6 +167,54 @@ void annotate_handle_request (flux_t *h,
     json_decref (tmp);
 }
 
+void annotate_memo_request (flux_t *h,
+                              flux_msg_handler_t *mh,
+                              const flux_msg_t *msg,
+                              void *arg)
+{
+    struct job_manager *ctx = arg;
+    struct flux_msg_cred cred;
+    flux_jobid_t id;
+    json_t *memo = NULL;
+    struct job *job;
+    const char *errstr = NULL;
+    int no_commit = 0;
+    json_t *tmp = NULL;
+
+    if (flux_request_unpack (msg, NULL, "{s:I s?b s:o}",
+                                        "id", &id,
+                                        "volatile", &no_commit,
+                                        "memo", &memo) < 0
+        || flux_msg_get_cred (msg, &cred) < 0)
+        goto error;
+    if (!(job = zhashx_lookup (ctx->active_jobs, &id))) {
+        errstr = "unknown job id";
+        errno = ENOENT;
+        goto error;
+    }
+    if (flux_msg_cred_authorize (cred, job->userid) < 0) {
+        errstr = "guests can only add a memo to their own jobs";
```

Comment:
s/add a memo to/memoize?

Reply:
I actually thought about calling it memoize, but [memoize](https://en.wikipedia.org/wiki/Memoization) already has a different definition, which might make the resulting error message confusing (I thought anyway)

Reply:
OK :-)  Your message is just fine and I should probably not have bothered you with that one.